### PR TITLE
Update JuMP syntax to use the new nonlinear interface

### PR DESCRIPTION
**Pull request by frapac**
I had [to comment out the tests](https://github.com/frapac/OptimizationProblems.jl/blob/fp/nljump/test/runtests.jl#L47-L53) checking that the formulation in JuMP matches those in ADNLPModels.

Apparently, the problem arises when we have `ScalarQuadraticFunction` in the MOI model. Maybe related to: 
https://github.com/JuliaSmoothOptimizers/NLPModelsJuMP.jl/pull/102

### CODE REVIEW

**Comment by tmigot** [comment]
Thanks a lot @frapac for the PR. NLPModelsJuMP still doesn't support the quadratic constraints, we lagged behind probably because we did not have a big usage of it. With the new nonlinear JuMP interface, I think we will try to add this quickly. I propose we wait a bit for this PR.

**Comment by gdalle** [comment]
@amontoison it might be a good time to revisit this PR based on what we discussed yesterday?

**Comment by amontoison** [comment]
Yes, I need to work on that since a long time.
I could have some bandwidth the end of next week.

**Comment by amontoison** [comment]
@frapac I released NLPModelsJuMP.jl 0.13.0.
We support `ScalarQuadraticFunction` now.

**Comment by amontoison** [comment]
@frapac @tmigot 
With the release 0.13.1 of NLPModels.jl, the new nonlinear API of JuMP is supported.
Benoît helped me to fix the last issue today during the Hackathon of JuMP-dev 2024.

File: test/runtests.jl [hunk 1]
```diff
@@ -44,13 +44,13 @@ include("test_utils.jl")
...
-    nlp_jump = MathOptNLPModel(model)
-    test_compatibility(prob, nlp_jump, nlp_ad, ndef)
-  end
+  # @testset "Test problems compatibility for $prob" begin
+  #   prob == :hs61 && continue #because nlpmodelsjump is not working here https://github.com/JuliaSmoothOptimizers/NLPModelsJuMP.jl/issues/84
+  #   prob_fn = eval(Meta.parse("PureJuMP.$(prob)"))
+  #   model = prob_fn(n = ndef)
+  #   nlp_jump = MathOptNLPModel(model)
+  #   test_compatibility(prob, nlp_jump, nlp_ad, ndef)
+  # end
```
Comment (tmigot) [inline test/runtests.jl:53]:
```suggestion
  @testset "Test problems compatibility for $prob" begin
    prob_fn = eval(Meta.parse("PureJuMP.$(prob)"))
    model = prob_fn(n = ndef)
    nlp_jump = MathOptNLPModel(model)
    test_compatibility(prob, nlp_jump, nlp_ad, ndef)
  end
```

**Review by tmigot — CHANGES_REQUESTED** [review]

**Comment by tmigot** [comment]
Superseded by #339

**Pull request closed without merging**